test/common_distributed.py

```diff
@@ -129,26 +129,39 @@ def setUp(self):
         super(MultiProcessTestCase, self).setUp()
         self.skip_return_code_checks = []
         self.rank = self.MAIN_PROCESS_RANK
-        self.file = tempfile.NamedTemporaryFile(delete=False)
+        # spawn can not pass file handle, so pass file name here
+        self.file = tempfile.NamedTemporaryFile(delete=False).name
```

Comment:
Looks like this PR has changes from the 'spawn' PR. Can we export using `ghexport` to avoid this?